db/cache/charts.go

```diff
@@ -479,24 +485,25 @@ func (charts *ChartData) Lengthen() error {
 		for _, interval := range intervals {
 			// For each new day, take an appropriate snapshot. Some sets use sums,
 			// some use averages, and some use the last value of the day.
+			days.Height = append(days.Height, uint64(interval[1]-1))
 			days.PoolSize = append(days.PoolSize, blocks.PoolSize.Avg(interval[0], interval[1]))
 			days.PoolValue = append(days.PoolValue, blocks.PoolValue.Avg(interval[0], interval[1]))
 			days.BlockSize = append(days.BlockSize, blocks.BlockSize.Sum(interval[0], interval[1]))
 			days.TxCount = append(days.TxCount, blocks.TxCount.Sum(interval[0], interval[1]))
 			days.NewAtoms = append(days.NewAtoms, blocks.NewAtoms.Sum(interval[0], interval[1]))
 			days.Chainwork = append(days.Chainwork, blocks.Chainwork[interval[1]])
 			days.Fees = append(days.Fees, blocks.Fees.Sum(interval[0], interval[1]))
-			days.Height = append(days.Height, uint64(interval[1]-1))
 		}
 	}
 
 	// Check that all relevant datasets have been updated to the same length.
-	daysLen, err := ValidateLengths(days.PoolSize, days.PoolValue, days.BlockSize,
-		days.TxCount, days.NewAtoms, days.Chainwork, days.Fees, days.Height, days.Time)
+	daysLen, err := ValidateLengths(days.Height, days.Time, days.PoolSize,
+		days.PoolValue, days.BlockSize, days.TxCount, days.NewAtoms,
+		days.Chainwork, days.Fees)
 	if err != nil {
 		return fmt.Errorf("day bin: %v", err)
 	} else if daysLen == 0 {
-		return fmt.Errorf("unexpected zero-length day-binned data")
+		log.Warnf("(*ChartData).Lengthen: Zero-length day-binned data!")
```

Comment:
@buck54321 With 199 blocks in the test DB, we don't get 1 day.  Is this OK?

Reply:
With a zero-based mainnet test db, we need 334 blocks to get a single day point, and 691 to get 2. 

I'm okay with a bigger test db if you are. Or we could explore artificially adding time to the block times to keep things small.

Reply:
Moving forward on master (after 5.0), I'd probably go with a bigger db, 2 days sounds good.

For getting the tests working for 5.0 release, do you see an issue with having this day-binned data error become a warning log?  It could be a legit error of course, when there really is 1 or more full days, but it seems like it should not be an error in this case where there just aren't that many blocks yet.

Reply:
I think that's the right move.